src/ts/message.ts

```diff
@@ -201,8 +207,8 @@ export function printMessage(fileName: string, exportMap: ExportMap, messageDesc
         }
         const fieldObjectName = normaliseFieldObjectName(camelCaseName);
         toObjectType.printIndentedLn(`${fieldObjectName}${canBeUndefined ? "?" : ""}: ${fieldObjectType},`);
-        printer.printIndentedLn(`get${withUppercase}(): ${exportType}${canBeUndefined ? " | undefined" : ""};`);
-        printer.printIndentedLn(`set${withUppercase}(value${type === MESSAGE_TYPE ? "?" : ""}: ${exportType}): void;`);
+        printer.printIndentedLn(`get${jsGetterName(withUppercase)}(): ${exportType}${canBeUndefined ? " | undefined" : ""};`);
+        printer.printIndentedLn(`set${jsGetterName(withUppercase)}(value${type === MESSAGE_TYPE ? "?" : ""}: ${exportType}): void;`);
```

Comment:
I tried to also verify if field presence is enabled, but the TS generation fails. It is possible there might be conflicts with additional methods (hasExtension, etc.) when it is supported.

Reply:
I think you were hitting [this issue](https://github.com/improbable-eng/ts-protoc-gen/pull/276) which is now fixed on `master`.

I've added the presence checks in [this commit](https://github.com/improbable-eng/ts-protoc-gen/commit/d4822f87a1d7ba0bda47e8e64ef5a6aeefeddf5a#diff-d0efdb75df2ff73ffd47aefae54c6e34b04e13e1cf6819e34e4da3b2b2b7219aL1). Can you add them to this PR please?

Reply:
I've rebased to master and pulled in that change - thanks!